src/boot/boot.c

```diff
@@ -1783,13 +1783,13 @@ static void config_select_default_entry(Config *config) {
         assert(config);
 
         i = config_find_entry(config, config->entry_oneshot);
-        if (i != IDX_INVALID) {
+        if (i != IDX_INVALID || config->entries[i]->tries_left != 0) {
```

Comment:
is this really the right approach? in particular for the oneshot stuff this means that if the user selects a specific entry for the next boot (maybe they just want to see the error the boot is failing with again) we'd ignore the request. which doesn't seem right.